dali/imgcodec/decoders/jpeg/jpeg_mem.cc

```diff
@@ -518,10 +499,8 @@ uint8* UncompressLow(const void* srcdata, FewerArgsForCompiler* argball) {
 //  parameters won't get clobbered by the longjmp.  So we help
 //  it out a little.
 uint8* Uncompress(const void* srcdata, int datasize,
-                  const UncompressFlags& flags, int64* nwarn,
-                  std::function<uint8*(int, int, int)> allocate_output) {
-  FewerArgsForCompiler argball(datasize, flags, nwarn,
-                               std::move(allocate_output));
+                  const UncompressFlags& flags) {
+  FewerArgsForCompiler argball(datasize, flags);
   uint8* const dstdata = UncompressLow(srcdata, &argball);
```

Comment:
So here we releasded the ownership of the memory and the caller of the function must take care of releasing it, right? As a food for thought, maybe this function should also return smart pointer?